### PR TITLE
UI Input Password, Constraints

### PR DESCRIPTION
**Pull request by nhaagen**
This adds constraints to the password-input to enforce lower/upper/special chars and numbers.

### CODE REVIEW

File: src/UI/Implementation/Component/Input/Field/Password.php [hunk 1]
```diff
@@ -42,4 +44,43 @@ protected function isClientSideValueOk($value) {
...
+		$this->checkBoolArg('lower', $lower);
+		$this->checkBoolArg('upper', $upper);
+		$this->checkBoolArg('numbers', $numbers);
+		$this->checkBoolArg('special', $special);
+
+		$data = new \ILIAS\Data\Factory();
+		$validation = new \ILIAS\Validation\Factory($data);
+		$pw_validation = $validation->password();
+		$constraints = [
+            $pw_validation->hasMinLength($min_length),
```
Comment (klees) [inline src/UI/Implementation/Component/Input/Field/Password.php]: Indentation please.

File: src/UI/Component/Input/Field/Factory.php [hunk 2]
```diff
@@ -206,7 +206,7 @@ public function dependantGroup(array $inputs);
 	 */
 	public function checkbox($label, $byline = null);
 
-		/**
+	/**
```
Comment (klees) [inline src/UI/Component/Input/Field/Factory.php]: :+1:

**Review by klees — CHANGES_REQUESTED** [review]
I like very much. Tests for constraints are missing. Will still take this to JF.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 16 JUL 2018 : New rules are accepted by JF

**Comment by klees** [comment]
@Amstutz: Wanna have a look into this?

**Review by klees — APPROVED** [review]

**Comment by Amstutz** [comment]
@nhaagen Thx a lot for this contribution!

**Pull request merged**